### PR TITLE
[1.0] Fix for gatsby develop hiccuping on gatsby-ssr.js using `import`

### PR DESCRIPTION
**Pull request by mericsson**
@KyleAMathews this doesn't feel right given what you wrote in [PR 1081](https://github.com/gatsbyjs/gatsby/pull/1081#issuecomment-306925029) but I wasn't sure how to apply your comment. This workaround seems to work, but I'm not sure if it has downstream effects. Still acquainting myself with the code but thought I'd put this for review to get more ideas / feedback and collaborate on this issue.

The issue, to restate it, is when using ES6 features that require webpack like `import` in `gatsby-ssr.js` then running `gatsby develop` dies a quick death.

```
(function (exports, require, module, __filename, __dirname) { import React from 'react'
                                                              ^^^^^^
SyntaxError: Unexpected token import
    at Object.exports.runInThisContext (vm.js:76:16)
    at Module._compile (module.js:542:28)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/mericsson/dev/gatsby-using-redux/.cache/api-runner-ssr.js:2:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
```

Appreciate any help to dig into this. Thanks!

### CODE REVIEW

File: examples/redux/gatsby-ssr.js [hunk 1]
```diff
@@ -1,20 +1,16 @@
 import React from 'react'
-import { renderToString } from 'react-dom/server'
 import { Provider } from 'react-redux'
 
 import createStore from './src/state/createStore'
 
-exports.replaceServerBodyRender = ({ component: body }) => {
+exports.replaceRenderer = ({ bodyComponent, replaceBodyHTMLString }) => {
```
Comment (mericsson) [inline examples/redux/gatsby-ssr.js]: Applying latest APIs from here: https://www.gatsbyjs.org/docs/ssr-apis/#replaceRenderer

**Comment by KyleAMathews** [comment]
Yeah... this means any plugin which implements gatsby-ssr.js won't work as expected in development since they can't set their header/postbody components.

**Comment by mericsson** [comment]
Ah makes sense. OK feel free to close. Not familiar enough with webpack to know where to start with your suggestion. :/

**Comment by KyleAMathews** [comment]
The redux example changes would still be great to get in!

**Comment by mericsson** [comment]
OK I'll close this and open an new one with the redux example changes.

File: examples/redux/gatsby-ssr.js (hunk 1 above)
Comment (scottyeck) [inline examples/redux/gatsby-ssr.js]: Cool!

**Comment by scottyeck** [comment]
@mericsson Correct me if I'm wrong, but this was closed even though the original issue wasn't fixed - am I reading this right? If so, can we re-open? I'd love to get this fixed / fix it if need be. 😎

**Comment by mericsson** [comment]
@scottyeck yep you're right. This was closed though original issue was not fixed. I'll re-open it. If you want to give a try to fix it that would be awesome!

**Comment by KyleAMathews** [comment]
haha no it should actually be closed and an issue be created for this.

The normal pattern is:

1) create issue
2) create PR to fix issue

if PR doesn't work, repeat with more PRs until one is merged and then close issue 1.

The problem here is there wasn't an issue ever created so closing this PR left us without something tracking the issue.

Could one of you create an issue for this and we move on from there?

**Comment by mericsson** [comment]
I'll create an issue. thanks @KyleAMathews

**Comment by mericsson** [comment]
Opened [Issue 1156](https://github.com/gatsbyjs/gatsby/issues/1156) to track this.

**Comment by scottyeck** [comment]
Ah great - totally didn't realize this was a PR and not an issue. My bad.

Thanks all 😎

**Pull request closed without merging**